mlir/lib/Dialect/Bufferization/Transforms/BufferResultsToOutParams.cpp

```diff
@@ -102,6 +102,14 @@ updateFuncOp(func::FuncOp func,
   return success();
 }
 
+static bool isFunctionArgument(mlir::Value value) {
+  // Check if the value is a Function argument
+  if (auto blockArg = dyn_cast<mlir::BlockArgument>(value)) {
+    return blockArg.getOwner()->isEntryBlock();
+  }
+  return false;
+}
```

Comment:
This is not enough...  Entry block doesnt have to be func op's entry.

Reply:
So is there a way to check if a value is a func arg?

Reply:
Maybe you can check the parent op of entry block is func op

Reply:
Thanks for the suggestion, will change that.